### PR TITLE
comments support in reader, initial version

### LINKED ISSUE #653: Get_comments
I have loaded an xlsx file into memory and am attempting to load the comments attached to the file.

    DATA:
      lo_iterator         TYPE REF TO cl_object_collection_iterator,
      lo_comments         TYPE REF TO zcl_excel_comments,
      lo_comment          TYPE REF TO zcl_excel_comment,
      lo_excel_worksheet  TYPE REF TO zcl_excel_worksheet.

      ls_sheet_name = ls_range-sheet.
      lo_excel_worksheet = go_excel->get_worksheet_by_name( ls_sheet_name ) ##NO_TEXT.
      lo_iterator = lo_excel_worksheet->get_comments_iterator( ).
      WHILE lo_iterator->has_next( ) = abap_true.
          lo_comment ?= lo_iterator->get_next( ).
      ENDWHILE.

The excel file has comments in it, however has_next never evaluates to true. Am I doing something wrong, or is the functionality not present?

Thanks,
Jared

### PR DESCRIPTION
**Pull request by AndreaBorgia-Abo**
fix #653

### CODE REVIEW

**Comment by AndreaBorgia-Abo** [comment]
@larshp before un-drafting, I'd like your opinion on the linter message: 
`Method "get_attribute_node_ns" not found, methodCallChain `

The call was reworked because of a previous error for 702 compatibility, related to "CAST if_ixml_element" in that call.

**Comment by sandraros** [comment]
@AndreaBorgia-Abo You need to do a PR to add the signature of `get_attribute_node_ns` in abaplint/deps of `if_ixml_element`: https://github.com/abaplint/deps/blob/main/src/ixml/if_ixml_element.intf.abap

**Comment by AndreaBorgia-Abo** [comment]
[Done](https://github.com/abaplint/deps/pull/136), thanks.

**Review by AndreaBorgia-Abo — COMMENTED** [review]
changes are as expected

**Comment by AndreaBorgia-Abo** [comment]
found an edge case, commit incoming

**Comment by sandraros** [comment]
I'll check your PR tomorrow, and when it's no more a draft.

**Review by AndreaBorgia-Abo — COMMENTED** [review]
this should be it

File: src/zcl_excel_reader_2007.clas.abap [hunk 1]
```diff
@@ -3573,6 +3587,56 @@ CLASS zcl_excel_reader_2007 IMPLEMENTATION.
...
+
+      CLEAR lv_comment_text.
+      lo_attr = lo_node_comment->get_attribute_node_ns( name = 'ref' ).
+      lv_attr_value  = lo_attr->get_value( ).
+
+      lo_node_comment_child ?= lo_node_comment->get_first_child( ).
+      IF lo_node_comment_child IS BOUND.
+        lv_tag_name = lo_node_comment_child->get_name( ).
+        IF lv_tag_name = 't'. "simple strings
+          lv_comment_text = lo_node_comment_child->get_value( ).
```
Comment (sandraros) [inline src/zcl_excel_reader_2007.clas.abap]: This condition is always false in my tests (tag name is `text`), is there a test case for it? The rest is okay for me.

Comment (AndreaBorgia-Abo) [reply src/zcl_excel_reader_2007.clas.abap]: Yup, that code path was untested: using "Ecma Office Open XML Part 1 - Fundamentals And Markup Language Reference" (par.18.7) I could not decide whether pure text ("t" node alone) was allowed or not, so I coded that case as well. The dual t / rPr handling was loosely modeled after shared strings.

ZDEMO_EXCEL_COMMENTS creates the following xl/comments1.xml, which works just fine in Excel 2013:
`<?xml version="1.0"  standalone="yes"?><comments xmlns="http://schemas.openxmlformats.org/spreadsheetml/2006/main"><authors><author>DEVELOPER</author></authors><commentList><comment ref="B13" authorId="0"><text><r><rPr><b/><sz val="9"/><color indexed="81"/><rFont val="Tahoma"/><family val="2"/></rPr><t xml:space="preserve">This is how it begins to be debug time...</t></r></text></comment><comment ref="C18" authorId="0"><text><r><rPr><b/><sz val="9"/><color indexed="81"/><rFont val="Tahoma"/><family val="2"/></rPr><t xml:space="preserve">Another comment</t></r></text></comment></commentList></comments>`

If I remove the "rPr" node, then the re-zipped xlsx is no longer accepted:
`<?xml version="1.0"  standalone="yes"?><comments xmlns="http://schemas.openxmlformats.org/spreadsheetml/2006/main"><authors><author>DEVELOPER</author></authors><commentList><comment ref="B13" authorId="0"><text><r><t xml:space="preserve">This is how it begins to be debug time...</t></r></text></comment><comment ref="C18" authorId="0"><text><r><rPr><b/><sz val="9"/><color indexed="81"/><rFont val="Tahoma"/><family val="2"/></rPr><t xml:space="preserve">Another comment</t></r></text></comment></commentList></comments>`

I suppose we can safely assume comments always come in RTF and accept the fact we're only reading the text component.
Commit incoming...

**Review by AndreaBorgia-Abo — COMMENTED** [review]
changes are as expected

File: src/zcl_excel_reader_2007.clas.abap (hunk 1 above)
Comment (sandraros) [reply src/zcl_excel_reader_2007.clas.abap]: Thanks for clarifying!

**Review by sandraros — APPROVED** [review]

**Pull request merged**